core/src/nucleus/ribosome/runtime.rs

```diff
@@ -54,34 +55,33 @@ impl Runtime {
         String::from_utf8(bin_arg)
             // @TODO don't panic in WASM
             // @see https://github.com/holochain/holochain-rust/issues/159
-            .unwrap()
+            .unwrap().into()
     }
 
-    /// Store a string in wasm memory.
+    /// Store anything that implements Into<JsonString> in wasm memory.
+    /// Note that From<T> for JsonString automatically implements Into
     /// Input should be a a json string.
     /// Returns a Result suitable to return directly from a zome API function, i.e. an encoded allocation
-    pub fn store_utf8(&mut self, json_str: &str) -> ZomeApiResult {
+    pub fn store_as_json_string<J: Into<JsonString>>(&mut self, jsonable: J) -> ZomeApiResult {
+        let j: JsonString = jsonable.into();
         // write str to runtime memory
-        let mut s_bytes: Vec<_> = json_str.to_string().into_bytes();
+        let mut s_bytes: Vec<_> = j.into_bytes();
         s_bytes.push(0); // Add string terminate character (important)
 
-        let allocation_of_result = self.memory_manager.write(&s_bytes);
-        if allocation_of_result.is_err() {
-            return Err(Trap::new(TrapKind::MemoryAccessOutOfBounds));
+        match self.memory_manager.write(&s_bytes) {
+            Err(_) => ribosome_error_code!(NotAnAllocation),
+            Ok(allocation) => Ok(Some(RuntimeValue::I32(allocation.encode() as i32))),
         }
-        let encoded_allocation = allocation_of_result.unwrap().encode();
-
-        // Return success in i32 format
-        Ok(Some(RuntimeValue::I32(encoded_allocation as i32)))
     }
 
-    // Stack any Serializable data as a json for the Zome to read as a returned value.
-    pub fn store_as_json<T: serde::Serialize>(&mut self, data: T) -> ZomeApiResult {
-        let maybe_json = serde_json::to_string(&data);
-        match maybe_json {
-            Err(_) => ribosome_error_code!(ResponseSerializationFailed),
-            Ok(json) => self.store_utf8(&json),
-        }
+    pub fn store_result<J: Into<JsonString>>(
+        &mut self,
+        result: Result<J, HolochainError>,
+    ) -> ZomeApiResult {
+        self.store_as_json_string(match result {
+            Ok(address) => ZomeApiInternalResult::success(address),
```

Comment:
this makes it seem like the content of the the result OK(J) will always be an address, but from what I can tell it could be anything thats jsonifyable, not just an address.  Shouldn't we change the variable name to `x` or `content`?